### ISSUE TITLE
Benchmarks

### ISSUE THREAD

**Issue by ritchie46** [pola-rs/valves#6]
As we compare different tools here. It would be cool to run benchmarks from this repo.

Maybe in CI, and later maybe even a dedicated runner.

These can could then be shown on the website. I am already assuming here that polars does great. :smile:

**Comment by koaning** [comment]
Benchmarks are most certainly the plan! Any preference on how though? Part of me likes the idea of running it on Github Actions, but I'm wondering if they provide consistent hardware. There's also "what datasets shall we use" and "where to host those". I may also imagine that we may want to consider versions of functions. After all, there may be multiple ways to implement "sessionize".

**Comment by koaning** [comment]
Come to think of it, do we really want to download large datasets and run potentially long-running benchmarks in Github Actions?

**Comment by ritchie46** [comment]
Could create CI that only runs on manual triggers. In the polars repo, we create the datasets instead of downloading.

The VM's are shared, but I do think that within a pipeline we have the same compute (not really sure though), this would make relative comparisons still sensbible within one run.

**Comment by koaning** [comment]
Fair enough. Let's try and start with GithubCI just to keep things simple. 

Where would you want to store the data from the benchmark results? Do we want to store the results of the runs in git?

**Comment by ritchie46** [comment]
Hmm.. that's maybe a good idea yes. We could store it in a separate clean branch. The whole benchmarking is a large todo still. 

I also want to run TCPH benchmarks in the polars repo, which would need dedicated compute. I can imagine eventually setting up a database etc.

**Comment by koaning** [comment]
I just got a base thing goin' on my local multiple dispatch branch. 

![image](https://user-images.githubusercontent.com/1019791/144128175-e186f15c-652c-4608-9666-effb828eedd3.png)

What does TCPH stand for? 

Also, simulating some of these datasets is tricky. How might we properly simulate a session dataset?

**Comment by koaning** [comment]
I was thinking about building a [memo script](https://koaning.io/posts/for-loop-memo/), but I'm open to other ideas too. It kind of depends on how accurate you'd like these numbers. There's also stuff like measure parquet vs. csv and/or number of CPUs.
